src/helpers/api/apiAjax.js

```diff
@@ -15,14 +15,12 @@ class ApiAjax {
         const url = formatUrl(path);
 
         req.onload = () => {
-          if (req.status === 500) {
-            reject(req.response);
-            return;
+          if (req.status !== 200) {
```

Comment:
Yes, we need to replace this dummy Ajax implementation by `Zepto` or `fetch-please`. I checked `fetch-please` code and I liked. It has `abort` method which is cool when you want to stop a request in progress.